### DISCUSSION TITLE
Error completing request

### DISCUSSION THREAD

**Discussion by frame73s** [thygate/stable-diffusion-webui-depthmap-script#287] [General]
Hi  i cannot generate any 3d inpainted mesh because the script stops; this is my log:

DepthMap v0.3.12 (54193c4f)
device: cuda
Loading model weights from  ./models/leres/res101.pth
initialize network with normal
loading the model from ./models/pix2pix\latest_net_G.pth
Computing depthmap(s) ..
  0%|                                                                                            | 0/1 [00:00<?, ?it/s]

wholeImage being processed in : 1344
Adjust factor is: 1.0
Selecting patches ...
Target resolution:  (1764, 2688, 3)
Resulting depthmap resolution will be : (336, 512)
patches to process: 16
         processing patch 0 / 15 | [ 64   0 299 299]
         processing patch 1 / 15 | [128   0 299 299]
         processing patch 2 / 15 | [192   0 299 299]
         processing patch 3 / 15 | [  9   9 280 280]
         processing patch 4 / 15 | [ 27  91 244 244]
         processing patch 5 / 15 | [ 91  91 244 244]
         processing patch 6 / 15 | [155  91 244 244]
         processing patch 7 / 15 | [219  91 244 244]
         processing patch 8 / 15 | [299  43 213 213]
         processing patch 9 / 15 | [299 107 213 213]
         processing patch 10 / 15 | [  0  64 171 171]
         processing patch 11 / 15 | [  0 128 171 171]
         processing patch 12 / 15 | [128   0 171 171]
         processing patch 13 / 15 | [192   0 171 171]
         processing patch 14 / 15 | [256   0 171 171]
         processing patch 15 / 15 | [320   0 171 171]
  0%|                                                                                            | 0/1 [00:16<?, ?it/s]
*** Error completing request
*** Arguments: (0, <PIL.Image.Image image mode=RGB size=512x336 at 0x24A05076E00>, None, '', '', 0, 0, 512, 512, False, True, False, False, 0, 1, False, 1, True, True, False, False, ['left-right', 'red-cyan-anaglyph'], 2.5, 0, 'polylines_sharp', 0, True, False, False, False, False, 'u2net', False, 0, 2, False, None, True, False, True, False) {}
    Traceback (most recent call last):
      File "P:\stable-diffusion\modules\call_queue.py", line 55, in f
        res = list(func(*args, **kwargs))
      File "P:\stable-diffusion\modules\call_queue.py", line 35, in f
        res = func(*args, **kwargs)
      File "P:\stable-diffusion\extensions\stable-diffusion-webui-depthmap-script\scripts\depthmap.py", line 1236, in run_generate
        outputs, mesh_fi, meshsimple_fi = run_depthmap(
      File "P:\stable-diffusion\extensions\stable-diffusion-webui-depthmap-script\scripts\depthmap.py", line 727, in run_depthmap
        if not ('image has wrong mode' in str(ve) or 'I;16' in str(ve)): raise ve
      File "P:\stable-diffusion\extensions\stable-diffusion-webui-depthmap-script\scripts\depthmap.py", line 725, in run_depthmap
        images.save_image(Image.fromarray(img_output), path=outpath, basename=basename, seed=None, prompt=None, extension=opts.samples_format, info=info, short_filename=True,no_prompt=True, grid=False, pnginfo_section_name="extras", existing_info=None, forced_filename=None)
      File "P:\stable-diffusion\modules\images.py", line 578, in save_image
        os.makedirs(path, exist_ok=True)
      File "C:\Program Files\Python310\lib\os.py", line 225, in makedirs
        mkdir(name, mode)
    FileNotFoundError: [WinError 3] Impossibile trovare il percorso specificato: ''

---
Traceback (most recent call last):
  File "P:\stable-diffusion\venv\lib\site-packages\gradio\routes.py", line 422, in run_predict
    output = await app.get_blocks().process_api(
  File "P:\stable-diffusion\venv\lib\site-packages\gradio\blocks.py", line 1326, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "P:\stable-diffusion\venv\lib\site-packages\gradio\blocks.py", line 1229, in postprocess_data
    self.validate_outputs(fn_index, predictions)  # type: ignore
  File "P:\stable-diffusion\venv\lib\site-packages\gradio\blocks.py", line 1204, in validate_outputs
    raise ValueError(
ValueError: An event handler (f) didn't receive enough output values (needed: 5, received: 3).
Wanted outputs:
    [gallery, textbox, model3d, html, html]
Received outputs:
    [None, "", "<div class='error'>FileNotFoundError: [WinError 3] Impossibile trovare il percorso specificato: &#x27;&#x27;</div><div class='performance'><p class='time'>Time taken: <wbr>20.91s</p><p class='vram'>Torch active/reserved: 5339/6000 MiB, <wbr>Sys VRAM: 7949/12288 MiB (64.69%)</p></div>"]

**Comment by thygate** [comment]
```
File "P:\stable-diffusion\modules\images.py", line 578, in save_image
os.makedirs(path, exist_ok=True)
File "C:\Program Files\Python310\lib\os.py", line 225, in makedirs
mkdir(name, mode)
FileNotFoundError: [WinError 3] Impossibile trovare il percorso specificato: ''
```
It seems to be unable to save the images, or create the path to them .. 

I don't think it's raising this exception 
`if not ('image has wrong mode' in str(ve) or 'I;16' in str(ve)): raise ve`
but check you're not saving as jpg, but as png to be sure !

**Reply by frame73s** [reply]
i've checked i'm saving in .png
